wdl/pipelines/TechAgnostic/Visualization/SvQCPlots.wdl

```diff
@@ -0,0 +1,451 @@
+        samples: "List of sample names"
+        coverage_metrics: "List of coverage metrics for each sample"
+        callers: "List of SV callers"
+        reference_name: "Reference genome name"
+    }
+
+    input{
+        String gcs_vcf_dir
+        Array[String] samples
+        Array[Float] coverage_metrics
+        Array[String] callers
+        String reference_name = "GRCh38"
+    }
+
+    scatter(caller in callers){
+        scatter(sample in samples){
+            call bcfQuerySV{
+                input:
+                    sample_name = sample,
+                    input_vcf = gcs_vcf_dir + "/" + sample + "." + caller + ".vcf.gz",
+                    caller = caller,
+            }
+        }
+    }
+    Array[File] all_SV_stats = flatten(bcfQuerySV.all_SV_stat_out)
+
+    call concatSVstats{
+        input:
+            all_stats = all_SV_stats,
+            callers = callers,
+    }
+
+    call compileSVstats{
+        input:
+            samples = samples,
+            all_stats = all_SV_stats,
+            callers = callers,
+    }
+    call addCoverageToSVstats{
+        input:
+            coverage_stats = coverage_metrics,
+            samples = samples,
+            allStatsBySample = compileSVstats.allStatsBySample,
+            callers = callers,
+    }
+
+    call plotSVQCMetrics{
+        input:
+            all_stats_with_cov = addCoverageToSVstats.all_stats_with_cov,
+            all_stats_by_type = concatSVstats.all_stats_by_type,
+            callers = callers,
+            reference_name = reference_name,
+    }
+
+
+output{
+        Array[File] all_stats_by_type = concatSVstats.all_stats_by_type
+        Array[File] all_stats_with_cov = addCoverageToSVstats.all_stats_with_cov
+        Array[File] metric_plot_pdfs = plotSVQCMetrics.output_pdfs
+        File plot_notebook = plotSVQCMetrics.out_plot_single_sample_stats
+    }
+
+
+}
+
+
+task bcfQuerySV{
+
+    parameter_meta{
+        sample_name: "Sample name"
+        input_vcf: "Input vcf file"
+        caller: "SV caller used to generate input vcf file"
+    }
+
+    input{
+        String sample_name
+        File input_vcf
+        String caller
+        RuntimeAttr? runtime_attr_override
+    }
+
+    String sample_stat_out = sample_name + "." + caller + ".txt"
+
+    Int minimal_disk_size = (ceil(size(input_vcf, "GB") + 100 )) # 100GB buffer
+    Int disk_size = if minimal_disk_size > 100 then minimal_disk_size else 100
+
+    command{
+        set -euo pipefail
+
+        cat ~{input_vcf} | bcftools query -i '(INFO/SVLEN>49 || INFO/SVLEN<-49) && FILTER=="PASS"' --format "%SVTYPE\t%SVLEN\n" > ~{sample_stat_out}
+
+    }
+
+    output{
+        File all_SV_stat_out = sample_stat_out
+    }
+        #########################
+    RuntimeAttr default_attr = object {
+        cpu_cores:          2,
+        mem_gb:             8,
+        disk_gb:            disk_size,
+        boot_disk_gb:       10,
```

Comment:
can remove since docker is not that big.

Reply:
done